src/Avalonia.Base/Reactive/LightweightObservableBase.cs

```diff
@@ -168,6 +168,8 @@ protected void PublishNext(T value)
                     for(int i = 0; i < count; i++)
                     {
                         observers[i].OnNext(value);
+                        // Avoid memory leak by clearing the reference.
+                        observers[i] = null!;
                     }
 
                     ArrayPool<IObserver<T>>.Shared.Return(observers);
```

Comment:
Note: `clearArray: true` could be passed to `Return()` instead.
However, since the array contain references, it won't use the fast `Array.Clear` version anyway, so we might as well use the existing loop like you did.

As a bonus, the current implementation allows previous observers to be reclaimed early if a GC happens during the loop.